Miscellaneous fixes for PAPI rocm components

gcongiu opened the pull request:
Some environment variables are not set correctly for the rocm components in PAPI

gcongiu committed:
papi/rocm: fix path for ROCP_METRICS

ROCP_METRICS is set to the path containing the metrics.xml file.

gcongiu committed:
papi/rocm_smi: add PAPI_ROCMSMI_ROOT environment variable

PAPI_ROCMSMI_ROOT is set to the path to librocm-smi64.so

gcongiu commented:
@G-Ragghianti could you please have a look?

G-Ragghianti commented on var/spack/repos/builtin/packages/papi/package.py:
```diff
@@ -94,12 +94,13 @@ def setup_build_environment(self, env):
             env.append_flags("CFLAGS", "-I%s/rocprofiler/include" % spec["rocprofiler-dev"].prefix)
             env.set(
-                "ROCP_METRICS", "%s/rocprofiler/lib/metrics.xml" % spec["rocprofiler-dev"].prefix
+                "ROCP_METRICS", "%s/rocprofiler/metrics.xml" % spec["rocprofiler-dev"].prefix.lib
```
```suggestion
                "ROCP_METRICS", "%s/metrics.xml" % spec["rocprofiler-dev"].prefix.lib
```

gcongiu replied:
This won't work because for rocm < 5.2.0 the location is `<path_to_rocprofiler>/rocprofiler/lib`, while for rocm >= 5.2.0 the location is `<path_to_rocprofiler>/lib/rocprofiler`. I have added an `os.path.exists()` to use the right format accordingly.

G-Ragghianti commented on var/spack/repos/builtin/packages/papi/package.py:
```diff
@@ -93,13 +93,19 @@ def setup_build_environment(self, env):
-            env.set(
-                "ROCP_METRICS", "%s/rocprofiler/lib/metrics.xml" % spec["rocprofiler-dev"].prefix
-            )
+            if os.path.exists("%s/rocprofiler/metrics.xml" % spec["rocprofiler-dev"].prefix.lib):
```
Maybe simplify it with just this?
```suggestion
            env.set("ROCP_METRICS", find(spec["rocprofiler-dev"].prefix, "metrics.xml"))
```
I haven't yet tested this because the rocm stack takes so long to install.

gcongiu replied:
I only see string find() in python, which does not match the behavior described in your code. I could still write a `find_file` function that searches for a file inside a directory and returns it if found. However, this looks orthogonal to the build error. I think you are right. There might be a problem with PAPI master (specifically with the testing framework) that pops up with more recent versions of gcc. We had similar problems in the past when introducing testing with newer compiler versions.

gcongiu replied:
Another things that pops up is that your environment is using rocm-5.4.3, which is know to have some problem with the value of the counters reported. The latest known working version should be 5.3.0 but if you want to be safe I would go with 5.2.0 (which is the one I am using for my testing).

G-Ragghianti replied:
This is not really my environment.  This is what spack is determining is the correct versions to use based on what was requested and the papi package definition.  If papi doesn't work with a version of rocm, then that should be included as a `conflict()` in the spack package.  Otherwise, users would not know what the problem is.  The default that spack builds with when you do a simple `spack install papi+rocm` should work without customization.

gcongiu replied:
Yes, you are right. I meant the environment in the test system you referenced.

gcongiu replied:
> If papi doesn't work with a version of rocm, then that should be included as a `conflict()` in the spack package. Otherwise, users would not know what the problem is. The default that spack builds with when you do a simple `spack install papi+rocm` should work without customization.

I can update the PR to include the conflicting rocm versions

G-Ragghianti replied:
The `find()` is from spack in lib/spack/llnl/util/filesystem.py`.  I see it is used this way in other spack packages (e.g., openmpi).

> I only see string find() in python, which does not match the behavior described in your code. I could still write a `find_file` function that searches for a file inside a directory and returns it if found. However, this looks orthogonal to the build error.

G-Ragghianti replied:
> I meant the environment in the test system you referenced.

It isn't being affected by the environment.  That is something that spack generally tries hard to avoid.  We should assume that spack will try to use the newest version of any package that it can if there are no other constraints.  This is why we need conflicts() if papi is known to not work with certain dependency versions.  In this case:

```
conflicts("^hip@5.4:")
```

G-Ragghianti commented:
I have a test of the completed now.  It appears that there is a problem with the MOCP_METRICS contents being truncated when the string is too long.  See the end of this output: https://jenkins.icl.utk.edu/job/Users/job/Ragghianti/job/papi_spack/label=gpu_amd,testmode=build/3/console

G-Ragghianti commented:
I've set the Jenkins project to automatically rebuild if you push more changes to your branch.

gcongiu commented:
Hi Gerald,

I think the problem is that spack is installing the default PAPI, which is
still 6.0.0.1. The most recent version uses PATH_MAX for static string
arrays, so it should not overrun the array.

Can you try changing the job SPEC to use ***@***.*** instead?

Thank you,
Giuseppe

On Tue, Sep 12, 2023 at 7:57 PM G-Ragghianti ***@***.***>
wrote:

> I have a test of the completed now. It appears that there is a problem
> with the MOCP_METRICS contents being truncated when the string is too long.
> See the end of this output:
> https://jenkins.icl.utk.edu/job/Users/job/Ragghianti/job/papi_spack/label=gpu_amd,testmode=build/3/console
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spack/spack/pull/39860#issuecomment-1716182360>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGP5MTARRBVSN6ZXDULKG3X2CPA3ANCNFSM6AAAAAA4OOT5CE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

G-Ragghianti commented:
> I think the problem is that spack is installing the default PAPI, which is still 6.0.0.1. The most recent version uses PATH_MAX for static string arrays, so it should not overrun the array. Can you try changing the job SPEC to use papi@master instead?

OK, this is done, but the project is not building now.

gcongiu commented:
> > I think the problem is that spack is installing the default PAPI, which is still 6.0.0.1. The most recent version uses PATH_MAX for static string arrays, so it should not overrun the array. Can you try changing the job SPEC to use papi@master instead?
> 
> OK, this is done, but the project is having errors building now.

Weird. Now it's complaining about relocatable symbols in tests:

```
1479    /tmp/jenkins/workspace/Users/Ragghianti/papi_spack/db770b22/lib/sp
             ack/env/gcc/gcc -I../testlib -I../validation_tests -I.. -I. -I/tmp
             /jenkins/workspace/Users/Ragghianti/papi_spack/db770b22/opt/spack/
             __spack_path_placeholder__/__spack_path_placeholder__/__sp/linux-r
             ocky9-x86_64/gcc-11.4.0/rocprofiler-dev-5.4.3-gpvsn4w3xq7xgugbv34r
             evlhrffpsz52/rocprofiler/include -g -Wextra  -Wall -DROCM_PROF_ROC
             PROFILER -DHAVE_ROCM -DPAPI_NUM_COMP=5 -O1  -DSYSTEM system_child_
             overflow.c ../testlib/libtestlib.a ../libpapi.a  -o system_child_o
             verflow
  >> 1480    ld.lld: error: relocation R_X86_64_32 cannot be used against local
              symbol; recompile with -fPIC
     1481    >>> defined in ../../../testlib/libtestlib.a(test_utils.o)
     1482    >>> referenced by test_utils.c:38 (/tmp/jenkins/workspace/Users/Ra
             gghianti/papi_spack/db770b22/stage/spack-stage-papi-master-iie35iu
             d7kv7bp6dsbuwatmtpspd6fdi/spack-src/src/testlib/test_utils.c:38)
     1483    >>>               test_utils.o:(approx_equals) in archive ../../..
             /testlib/libtestlib.a
```
This might be a problem that comes up with newer compilers. I will test it locally to reproduce it and let you know.

gcongiu commented:
> > > I think the problem is that spack is installing the default PAPI, which is still 6.0.0.1. The most recent version uses PATH_MAX for static string arrays, so it should not overrun the array. Can you try changing the job SPEC to use papi@master instead?
> > 
> > 
> > OK, this is done, but the project is having errors building now.
> 
> Weird. Now it's complaining about relocatable symbols in tests:
> 
> ```
> 1479    /tmp/jenkins/workspace/Users/Ragghianti/papi_spack/db770b22/lib/sp
>              ack/env/gcc/gcc -I../testlib -I../validation_tests -I.. -I. -I/tmp
>              /jenkins/workspace/Users/Ragghianti/papi_spack/db770b22/opt/spack/
>              __spack_path_placeholder__/__spack_path_placeholder__/__sp/linux-r
>              ocky9-x86_64/gcc-11.4.0/rocprofiler-dev-5.4.3-gpvsn4w3xq7xgugbv34r
>              evlhrffpsz52/rocprofiler/include -g -Wextra  -Wall -DROCM_PROF_ROC
>              PROFILER -DHAVE_ROCM -DPAPI_NUM_COMP=5 -O1  -DSYSTEM system_child_
>              overflow.c ../testlib/libtestlib.a ../libpapi.a  -o system_child_o
>              verflow
>   >> 1480    ld.lld: error: relocation R_X86_64_32 cannot be used against local
>               symbol; recompile with -fPIC
>      1481    >>> defined in ../../../testlib/libtestlib.a(test_utils.o)
>      1482    >>> referenced by test_utils.c:38 (/tmp/jenkins/workspace/Users/Ra
>              gghianti/papi_spack/db770b22/stage/spack-stage-papi-master-iie35iu
>              d7kv7bp6dsbuwatmtpspd6fdi/spack-src/src/testlib/test_utils.c:38)
>      1483    >>>               test_utils.o:(approx_equals) in archive ../../..
>              /testlib/libtestlib.a
> ```
> 
> This might be a problem that comes up with newer compilers. I will test it locally to reproduce it and let you know.

I tried building PAPI from sources using gcc 11.2.0 (no spack) and it works. With spack I get some complain when installing hsa-rocr-dev (something concerning the rocm device library missing for gfx1100).

gcongiu commented:
@G-Ragghianti this PR is somewhat urgent. Is there anything problematic that pops to the eye with it? I think the errors you were experiencing might have something to do with the environment in the test system...

G-Ragghianti commented:
It is fine that you are able to build outside of spack, and this may provide some pointer to the problem.  However, this is a change specifically to get the spack package buildable, and we are so far not able to complete that objective.  I am currently doing a build within a clean docker container.  This should eliminate any possible problems with my test environment.  I will update you on how that goes in the next few hours.

gcongiu commented:
Sounds good. Thank you

G-Ragghianti commented:
I get the same error in the docker container.  https://jenkins.icl.utk.edu/job/Users/job/Ragghianti/job/papi_spack/8/label=docker_dopamine,testmode=build/console

I suspect that this isn't a reflection of the spack package functionality but of the current state of the master branch in papi.

Here is the full concretization: 

```
==> Concretized papi@master%gcc@11+rocm amdgpu_target=gfx90a
 -   t6k5erk  papi@master%gcc@11.3.1~cuda~debug+example~infiniband~lmsensors~nvml~powercap~rapl+rocm~rocm_smi~sde+shared~static_tools amdgpu_target=gfx90a build_system=autotools arch=linux-rocky9-x86_64
 -   t5egbc2      ^hip@5.4.3%gcc@11.3.1~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5068750,c2ee21c,ca523f1,ddd86f0 arch=linux-rocky9-x86_64
 -   ekt7coc          ^cmake@3.27.4%gcc@11.3.1~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rocky9-x86_64
 -   yv77f2k              ^curl@8.1.2%gcc@11.3.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-rocky9-x86_64
 -   zobfvo3                  ^nghttp2@1.52.0%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   w5stopt          ^comgr@5.4.3%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64
 -   thk235p              ^rocm-cmake@5.5.1%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64
 -   m5cfi6v                  ^rocm-core@5.5.1%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64
 -   5wejwlt          ^glx@1.4%gcc@11.3.1 build_system=bundle arch=linux-rocky9-x86_64
 -   3xhn5ff              ^mesa@23.0.3%gcc@11.3.1+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared arch=linux-rocky9-x86_64
 -   lgeliky                  ^bison@3.8.2%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   ie6bgkp                  ^flex@2.6.3%gcc@11.3.1+lex~nls build_system=autotools arch=linux-rocky9-x86_64
 -   lhzszj4                      ^findutils@4.9.0%gcc@11.3.1 build_system=autotools patches=440b954 arch=linux-rocky9-x86_64
 -   vye5m3m                  ^glproto@1.4.17%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   kkjq4ig                      ^util-macros@1.19.3%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   oes4hvp                  ^libunwind@1.6.2%gcc@11.3.1~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-rocky9-x86_64
 -   ese4snh                  ^libx11@1.8.4%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   odwk2vi                      ^inputproto@2.3.2%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   5goa5pw                      ^kbproto@1.0.7%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   jhypbcr                      ^xextproto@7.3.0%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   ce6thpz                      ^xproto@7.0.31%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   pmfadn5                      ^xtrans@1.4.0%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   ursvlje                  ^libxcb@1.14%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   7ynyn5t                      ^libxau@1.0.8%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   dx7zmwe                      ^libxdmcp@1.1.4%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   fikgkin                      ^xcb-proto@1.15.2%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   54rciwk                  ^libxext@1.3.3%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   l7h7sx2                  ^libxt@1.1.5%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   f3owuc6                      ^libice@1.0.9%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   bviegxc                      ^libsm@1.2.3%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   b3zn6kw                  ^llvm@14.0.6%gcc@11.3.1+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~z3 build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime patches=1f42874,25bc503,6379168,8248141,b216cff,cb8e645 shlib_symbol_version=none targets=all version_suffix=none arch=linux-rocky9-x86_64
 -   yomkqne                      ^binutils@2.41%gcc@11.3.1~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-rocky9-x86_64
 -   2geuhsy                      ^hwloc@2.9.1%gcc@11.3.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rocky9-x86_64
 -   wxrvpbt                      ^libedit@3.1-20210216%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   dvsix3o                      ^lua@5.3.6%gcc@11.3.1~pcfile+shared build_system=makefile fetcher=curl arch=linux-rocky9-x86_64
 -   ek2vdgo                          ^unzip@6.0%gcc@11.3.1 build_system=makefile arch=linux-rocky9-x86_64
 -   mhs445q                      ^perl-data-dumper@2.173%gcc@11.3.1 build_system=perl arch=linux-rocky9-x86_64
 -   ornariu                      ^swig@4.0.2-fortran%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   bs4tc6d                          ^pcre@8.45%gcc@11.3.1~jit+multibyte+utf build_system=autotools arch=linux-rocky9-x86_64
 -   z25ittv                  ^meson@1.2.0%gcc@11.3.1 build_system=python_pip patches=0f0b1bd arch=linux-rocky9-x86_64
 -   e6pkmoa                  ^py-mako@1.2.4%gcc@11.3.1 build_system=python_pip arch=linux-rocky9-x86_64
 -   hd7qefg                      ^py-markupsafe@2.1.3%gcc@11.3.1 build_system=python_pip arch=linux-rocky9-x86_64
 -   udlh55k                  ^xrandr@1.5.0%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   fgzrcqz                      ^libxrandr@1.5.3%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   qrbdgjf                          ^renderproto@0.11.1%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   jb4kmkp                      ^libxrender@0.9.10%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   g7w7ak7                      ^randrproto@1.5.0%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   twtljkx          ^gmake@4.4.1%gcc@11.3.1~guile build_system=autotools arch=linux-rocky9-x86_64
 -   igetfmo          ^hipify-clang@5.4.3%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make patches=fafe6a5 arch=linux-rocky9-x86_64
 -   sqlbdge          ^hsakmt-roct@5.4.3%gcc@11.3.1~ipo+shared build_system=cmake build_type=Release generator=make patches=114b05a arch=linux-rocky9-x86_64
 -   mrusntn              ^libdrm@2.4.115%gcc@11.3.1~docs build_system=generic arch=linux-rocky9-x86_64
 -   qkqcggv                  ^libpciaccess@0.17%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   pwakthk                  ^libpthread-stubs@0.4%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   ljb26zf          ^numactl@2.0.14%gcc@11.3.1 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rocky9-x86_64
 -   e6n5pi5              ^autoconf@2.69%gcc@11.3.1 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-rocky9-x86_64
 -   hjd3mce              ^automake@1.16.5%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   6szpfut              ^libtool@2.4.7%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   sm527dv              ^m4@1.4.19%gcc@11.3.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rocky9-x86_64
 -   5v7uj6w                  ^diffutils@3.9%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   riqpxd7                  ^libsigsegv@2.14%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   4dhsyri          ^perl@5.38.0%gcc@11.3.1+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-rocky9-x86_64
 -   vz6za2m              ^berkeley-db@18.1.40%gcc@11.3.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rocky9-x86_64
 -   xpnrdms              ^bzip2@1.0.8%gcc@11.3.1~debug~pic+shared build_system=generic arch=linux-rocky9-x86_64
 -   ogxq256              ^gdbm@1.23%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   y23iyt3          ^perl-file-which@1.27%gcc@11.3.1 build_system=perl arch=linux-rocky9-x86_64
 -   loayfji          ^perl-uri-encode@1.1.1%gcc@11.3.1 build_system=perl arch=linux-rocky9-x86_64
 -   udqm4dq              ^perl-module-build@0.4232%gcc@11.3.1 build_system=perl arch=linux-rocky9-x86_64
 -   fjjfblp          ^py-cppheaderparser@2.7.4%gcc@11.3.1 build_system=python_pip arch=linux-rocky9-x86_64
 -   jb7vhx6              ^py-pip@23.1.2%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   47glvns              ^py-ply@3.11%gcc@11.3.1 build_system=python_pip arch=linux-rocky9-x86_64
 -   rbdm5u4              ^py-setuptools@68.0.0%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   cph3v2s              ^py-wheel@0.37.1%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   2b6lflq          ^rocminfo@5.4.3%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64
 -   waqfau6          ^roctracer-dev-api@5.4.3%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   meax7kd      ^hsa-rocr-dev@5.4.3%gcc@11.3.1+image~ipo+shared build_system=cmake build_type=Release generator=make patches=71e6851 arch=linux-rocky9-x86_64
 -   mg3uu3n          ^elfutils@0.189%gcc@11.3.1~debuginfod+exeprefix+nls build_system=autotools arch=linux-rocky9-x86_64
 -   tbmrefu              ^gettext@0.21.1%gcc@11.3.1+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-rocky9-x86_64
 -   yt6uoeq                  ^libxml2@2.10.3%gcc@11.3.1+pic~python+shared build_system=autotools arch=linux-rocky9-x86_64
 -   bx5tcuy                  ^tar@1.34%gcc@11.3.1 build_system=autotools zip=pigz arch=linux-rocky9-x86_64
 -   6fbro7m                      ^pigz@2.7%gcc@11.3.1 build_system=makefile arch=linux-rocky9-x86_64
 -   634zeeg              ^libiconv@1.17%gcc@11.3.1 build_system=autotools libs=shared,static arch=linux-rocky9-x86_64
 -   mmhjkxv              ^xz@5.4.1%gcc@11.3.1~pic build_system=autotools libs=shared,static arch=linux-rocky9-x86_64
 -   oosvrns              ^zstd@1.5.5%gcc@11.3.1+programs build_system=makefile compression=none libs=shared,static arch=linux-rocky9-x86_64
 -   bnnjukn          ^pkgconf@1.9.5%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   ylwpg65          ^xxd-standalone@8.2.1201%gcc@11.3.1 build_system=makefile arch=linux-rocky9-x86_64
 -   fhyy6sw      ^llvm-amdgpu@5.4.3%gcc@11.3.1~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=a08bbe1 arch=linux-rocky9-x86_64
 -   bid4cgj          ^ncurses@6.4%gcc@11.3.1~symlinks+termlib abi=none build_system=autotools arch=linux-rocky9-x86_64
 -   7rxwvnc          ^ninja@1.11.1%gcc@11.3.1+re2c build_system=generic arch=linux-rocky9-x86_64
 -   mq5h6ud              ^re2c@2.2%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   wsbv7dk          ^python@3.11.4%gcc@11.3.1+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,f2fd060 arch=linux-rocky9-x86_64
 -   axhlgku              ^expat@2.5.0%gcc@11.3.1+libbsd build_system=autotools arch=linux-rocky9-x86_64
 -   lrcbbki                  ^libbsd@0.11.7%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   bzscysr                      ^libmd@1.0.4%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   egdjbbt              ^libffi@3.4.4%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   7qf2cix              ^libxcrypt@4.4.35%gcc@11.3.1~obsolete_api build_system=autotools patches=4885da3 arch=linux-rocky9-x86_64
 -   btxhrs7              ^openssl@3.1.2%gcc@11.3.1~docs+shared build_system=generic certs=mozilla arch=linux-rocky9-x86_64
 -   bpdunu4                  ^ca-certificates-mozilla@2023-05-30%gcc@11.3.1 build_system=generic arch=linux-rocky9-x86_64
 -   3bm7zqo              ^readline@8.2%gcc@11.3.1 build_system=autotools patches=bbf97f1 arch=linux-rocky9-x86_64
 -   ld6dj32              ^sqlite@3.42.0%gcc@11.3.1+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rocky9-x86_64
 -   z6vzn54              ^util-linux-uuid@2.38.1%gcc@11.3.1 build_system=autotools arch=linux-rocky9-x86_64
 -   47jaxhg          ^z3@4.11.2%gcc@11.3.1~gmp~ipo~python build_system=cmake build_type=Release generator=make arch=linux-rocky9-x86_64
 -   etbx4fx          ^zlib-ng@2.1.3%gcc@11.3.1+compat+opt build_system=autotools patches=299b958,ae9077a,b692621 arch=linux-rocky9-x86_64
 -   5uxjpvh      ^rocprofiler-dev@5.4.3%gcc@11.3.1~ipo build_system=cmake build_type=Release generator=make patches=16754a1,c482eee arch=linux-rocky9-x86_64

```

Commit:
papi/rocm: add conflicts for rocm@5.4 onwards

rocprofiler versions after 5.4.0 onword do not work correctly and should be
avoided.

gcongiu commented:
@G-Ragghianti R_X86_64_32 looks like a suspicious relocation type. Shouldn’t that be R_X86_64_64 for 64-bit systems?

gcongiu commented:
> @G-Ragghianti R_X86_64_32 looks like a suspicious relocation type. Shouldn’t that be R_X86_64_64 for 64-bit systems?

Nevermind, that should not cause problems for small executables.

alalazo commented:
Is there any update with respect to this PR?

gcongiu commented:
This has become stale. @adanalis any chance you can pick it up? It should be a matter or updating the range of working rocm versions.

alalazo commented:
@gcongiu Thanks for the prompt feedback. Ping me for a review when ready
